unstoppable-ios-app/domains-manager-ios/Services/Networking/NetworkService.swift

```diff
@@ -376,9 +369,9 @@ extension NetworkService {
     }
     
     /// This function will return UD/ENS/Null name and corresponding PFP if available OR throw 404
-    func fetchGlobalReverseResolution(for address: HexAddress) async throws -> GlobalRR? {
+    func fetchGlobalReverseResolution(for identifier: HexAddress) async throws -> GlobalRR? {
```

Comment:
Usually it's better to have either 'throw' or return nil, not both. Can this be improved?

Reply:
100% Agreed. In this particular case throws responsible for process itself and optional responsible for accurate response. 
For example when wallet address doesn't have any domain, request won't fail but response is nil.